app/helpers/automation_templates_helper.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module AutomationTemplatesHelper
+  # Finds the harvest report for a specific harvest definition in an automation run
+  # @param last_automation_run [Automation] The last automation run
+  # @param harvest_definition [HarvestDefinition] The harvest definition to find the report for
+  # @param position [Integer] The position of the step in the automation
+  # @return [HarvestReport, nil] The harvest report if found, nil otherwise
+  def find_harvest_report(last_automation_run, harvest_definition, position)
```

Comment:
What happens if someone edits the automation steps while this is running?

Reply:
users can't edit the steps once its run, only the template

Reply:
Noice :D